src/sql/tests/querymodel/lowering

```diff
@@ -829,3 +828,73 @@ select * from x full outer join y on x.a = y.b
 | Project (#0..=#4)
 ----
 ----
+
+lower
```

Comment:
Missing feature. Previously, there was no CTE handling. 

If your graph looks like this:
```
 A
/  \
B   C
\  /
  D
  |
  E
```

The lowering code will treat it like this:
```
 A
/  \
B   C
|   |
D   D-clone
|   |
E   E-clone
```

The resulting `MirRelationExpr` should still be correct, but it will be significantly longer than what you get from `EXPLAIN DECORRELATED PLAN`, and it will be annoying to compare the `MirRelationExpr` lowered from QGM to the one you get from `EXPLAIN DECORRELATED PLAN` for correctness.

Reply:
OK, great addition!